becquerel/core/fitting.py

```diff
@@ -0,0 +1,537 @@
+from __future__ import print_function
+import numpy as np
+import pandas as pd
+import scipy.special
+from lmfit.model import Model
+import matplotlib.pyplot as plt
+from matplotlib.gridspec import GridSpec
+from matplotlib.font_manager import FontProperties
+
+
+FWHM_SIG_RATIO = 2.35482
+
+
+class FittingError(Exception):
+    """Exception raised by Fitters."""
+    pass
+
+
+# -----------------------------------------------------------------------------
+# Base functions
+# -----------------------------------------------------------------------------
+
+
+def constant(x, c):
+    return np.ones_like(x) * c
+
+
+def line(x, m, b):
+    return m * x + b
+
+
+def gauss(x, amp, mu, sigma):
+    amp = np.abs(amp)
+    mu = np.abs(mu)
+    sigma = np.abs(sigma)
+    return amp / np.sqrt(2. * sigma ** 2. * np.pi) * \
+        np.exp(-(x - mu) ** 2. / (2. * sigma ** 2.))
+
+
+def erf(x, amp, mu, sigma):
+    return amp * 0.5 * (1. - scipy.special.erf((x - mu) / (1.414214 * sigma)))
+
+
+def exp(x, amp, lam):
+    return amp * np.exp(x / lam)
+
+
```

Comment:
For some reason, I can't find it in this diff, but there are some undefined names in `def expgaussian()` on what I've pulled to my machine.

Reply:
is this still the case?

Reply:
No, looks like all names are defined now.